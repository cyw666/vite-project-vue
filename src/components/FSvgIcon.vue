<template>
  <svg class="f-svg-icon" aria-hidden="true" :style="svgStyle">
    <use :xlink:href="symbolId" />
  </svg>
</template>

<script lang="ts">
export default {
  name: 'FSvgIcon',
}
</script>

<script setup lang="ts">
const props = withDefaults(
  defineProps<{
    name: string
    size?: number | string
    color?: string
  }>(),
  {
    size: 16,
    color: '#303133',
  }
)

const svgStyle = computed(() => ({
  fontSize: `${props.size}px`,
  color: `${props.color}`,
}))

const symbolId = computed(() => `#icon-${props.name}`)
</script>

<style>
.f-svg-icon {
  width: 1em;
  height: 1em;
  vertical-align: -0.15em;
  fill: currentColor;
  overflow: hidden;
}
</style>
