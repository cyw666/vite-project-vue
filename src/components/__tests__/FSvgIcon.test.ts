import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import FSvgIcon from '../FSvgIcon.vue'

describe('FSvgIcon', () => {
  const name = 'logo'
  it('render test', () => {
    const wrapper = mount(FSvgIcon, {
      props: {
        name,
      },
    })
    expect(wrapper.find('use').html()).toContain(`#icon-${name}`)
  })
  it('icon size', () => {
    const size = 48
    const wrapper = mount(FSvgIcon, {
      props: {
        name,
        size,
      },
    })
    expect(wrapper.find('svg').attributes('style')).toContain(
      `font-size: ${size}px;`
    )
  })

  it('icon color', () => {
    const color = '#333333'
    const wrapper = mount(FSvgIcon, {
      props: {
        name,
        color,
      },
    })
    expect(wrapper.find('svg')).toMatchSnapshot()
  })
})
