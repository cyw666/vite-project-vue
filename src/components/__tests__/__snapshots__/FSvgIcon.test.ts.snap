// Vitest Snapshot v1

exports[`FSvgIcon > icon color 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <svg
    aria-hidden="true"
    class="f-svg-icon"
    style="font-size: 16px; color: rgb(51, 51, 51);"
  >
    <use
      xlink:href="#icon-logo"
    />
  </svg>,
}
`;
